openid-4-verifiable-presentations-1_0.md

```diff
@@ -542,6 +542,266 @@ The Wallet then validates the request as specified in OAuth 2.0 [@RFC6749].
+    "city": "Milliways",
+    "postal_code": "12345"
+  },
+  "degrees": [
+    {
+      "type": "Bachelor of Science",
+      "university": "University of Betelgeuse"
+    },
+    {
+      "type": "Master of Science",
+      "university": "University of Betelgeuse"
+    }
+  ],
+  "nationalities": ["British", "Betelgeusian"]
+}
+```
+
+The following shows examples of claims path queries and the respective selected
+claims:
+
+- `["name"]`: The claim `name` with the value `Arthur Dent` is selected.
+- `["address"]`: The claim `address` with its sub-claims as the value is
+  selected.
+- `["address", "street_address"]`: The claim `street_address` with the value `42
+  Market Street` is selected.
+- `["degrees", "*", "type"]`: All `type` claims in the `degrees` array are
+  selected.
+- `["nationalities", 1]`: The second nationality is selected.
+- `[]` (empty array): The entire Credential is selected.
+
+#### Processing
+
+In detail, the array is processed by the Wallet from left to right as follows:
+
+ 1. Select the root element of the credential, i.e., the top-level JSON object.
+ 2. Process the query components from left to right:
+    1. If the query component is a string, select the element in the respective
+       key in the currently selected element(s). If any of the currently
+       selected element(s) is not an object, abort processing and return an
+       error. If the key does not exist in any element currently selected,
+       remove that element from the selection.
+    2. If the query component is `*`, select all elements of the currently
+       selected array(s). If any of the currently selected element(s) is not an
+       array, abort processing and return an error.
+    3. If the query component is a non-negative integer, select the element at
+       the respective index in the currently selected array(s). If any of the
+       currently selected element(s) is not an array, abort processing and
+       return an error. If the index does not exist in a selected array, remove
+       that array from the selection.
+  3. If the set of elements currently selected is empty, abort processing and
+     return an error.
+
+The result of the processing is the set of elements which is requested for
+presentation.
+
+
+## Purpose Description {#purpose_description}
+
+A purpose description array is an object that describes the purpose for which
+the Verifier is requesting the Verifiable Credential. It can contain multiple
+objects, each describing the purpose in a specific language. Each object
+contains the following properties:
+
+`lang:`
+: REQUIRED. A string that specifies the language of the purpose
+description. The value of this property MUST be a valid language tag as defined
+in [@!BCP47].
+
+`text`:
+: REQUIRED. A string that contains the purpose description in the
+language specified by the `lang` property.
+
+## Credential Rules {#credential_rules}
+
+Just as for claims, the Verifier can define rules that specify which
```

Comment:
We are using the PE submission_requirements feature. I see the intent here is to replace `submission_requirements` with `rules`. The features that we in HR use cases that are missing from rules are

1. ability to require `all` credentials of a particular credential type (eg. all transcripts) vs one of a particular type (eg mDL). From my understanding both would expressed as `{ "required": 1, "from": [{"ref": "transcript"}]}` and `{ "required": 1, "from": [{"ref": "mDL"}]}`
2. in HR use cases the user has more flexibility in what to share. therefore the ability to also require at least one is useful `{"min": 1, "from": [{"ref": "email"}]}`.
3. explaining the purpose of  a `rule` (e.g. the RP requires a mDL or national id card to get the applicants name)
4. is there a maximum depth to rules?

Reply:
> ability to require all credentials of a particular credential type (eg. all transcripts) vs one of a particular type (eg mDL). From my understanding both would expressed as { "required": 1, "from": [{"ref": "transcript"}]} and { "required": 1, "from": [{"ref": "mDL"}]}
in HR use cases the user has more flexibility in what to share. therefore the ability to also require at least one is useful {"min": 1, "from": [{"ref": "email"}]}.

As per a simiilar suggestion above `required` should be `all`, `optional`, `one`, `at_least_one` with the default either being `all` or `optional`. An alternative if a number is required for certain use cases would be to replace `one` with `count` which would then have an additional property with the expected `count`

Reply:
The `at_least_one` case is not currently captured, but we should consider introducing it and therefore moving back to keywords instead of numbers. Do you have a concrete use case in mind for `at_least_one`?

Reply:
How would a purpose for the rule be exposed to the user?

Reply:
@danielfett the current use case we have is in the application for a job. In the recruitment use case the holder can share whatever information they feel is relevant. For example, a person may have had studied and achieved a number of diplomas, but one or more maybe irrelevant and/or perhaps detrimental to them being successful in the application. The recruiter therefore will request at least one degree but leave it up to the holder to select the relevant one or two. 

Again the recruitment use case means that certain types of credentials are needed for a purpose. We have a UI in the reference wallet in the HR use case that takes the user through a wizard-like process to meet each rule requirement. A purpose for that rule is required for this UX to make sense